pkgs/by-name/in/inochi-session/package.nix

```
@@ -0,0 +1,148 @@
+  };
+
+  nativeBuildInputs = [
+    dub
+    ldc
+    cmake # used for building `i2d-imgui`
+    copyDesktopItems
+    makeWrapper
+
+    # A fake git implementation for the `gitver` package used by `inochi-session` and `inochi2d`
+    # This will cause inochi2d to have the same version as inochi-session, as both are using the same library
+    (writeShellScriptBin "git" "echo ${finalAttrs.src.rev}")
+  ];
+
+  buildInputs = [
+    dbus
+    freetype
+    SDL2
+  ];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME=$(mktemp -d)
+
+    # register downloaded dependencies
+    mkdir deps
+    ${builtins.concatStringsSep "\n" (map ({ pname, version, src }: ''
+      cp -r --no-preserve=all ${src} deps/${pname}
+      dub add-local deps/${pname} ${version}
+    '') deps)}
+
+    # As per request of the author (https://github.com/NixOS/nixpkgs/issues/170315)
+    # we are not allowed to package the logo and the mascot of the application
+    sed -i '/InBranding/d' dub.sdl
+
+    sed -i "/bindbc-spout2/d" dub.sdl # windows-only dependency
+    sed -i "/vibe-d/d" deps/facetrack-d/dub.sdl # behind the "web-adaptors" configuration of `facetrack-d`
+    sed -i "/dunit/d" deps/ddbus/dub.json # would be used by the tests of `ddbus`
+
+    cp -r --no-preserve=all ${cimgui-src}/* deps/i2d-imgui/deps/cimgui
+    sed -i deps/i2d-imgui/deps/CMakeLists.txt \
+        -e "/PullSubmodules/d" \
+        -e "s/\''${cimgui_SUBMOD_DIR}/cimgui/g"
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    dub build --skip-registry=all --config=meta # uses the fake git
+    dub build --skip-registry=all --build=release --config=linux-full
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 LICENSE out/*LICENSE* -t $out/share/doc
+    install -Dm755 out/*.so -t $out/lib
+    install -Dm755 out/inochi-session -t $out/bin
+
+    runHook postInstall
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "inochi-session";
+      exec = "inochi-session";
+      desktopName = "Inochi Session";
+      comment = finalAttrs.meta.description;
+      categories = [ "Utility" ];
+    })
+  ];
+
+  preFixup = ''
+    patchelf $out/bin/inochi-session --add-needed cimgui.so
+  '';
+
+  dontStrip = true; # symbol lookup error: undefined symbol: , version
+
+  # open file dialog support
+  postFixup = ''
+    wrapProgram $out/bin/inochi-session \
+        --prefix PATH : ${lib.makeBinPath [ gnome.zenity ]}
+  '';
+
+  meta = {
+    changelog = "https://github.com/Inochi2D/inochi-session/releases/tag/${finalAttrs.src.rev}";
+    description = "An application that allows streaming with Inochi2D puppets";
+    homepage = "https://inochi2d.com/";
+    license = lib.licenses.bsd2;
+    mainProgram = "inochi-session";
+    maintainers = with lib.maintainers; [ tomasajt ];
+    platforms = lib.platforms.linux; # darwin is supported by upstream, but has slightly different build steps
```

In this case I'd prefer

```suggestion
    platforms = with lib.platforms; linux ++ darwin;
    broken = stdenv.isDarwin; # darwin is supported by upstream, but has slightly different build steps
```